src/coreclr/jit/valuenum.cpp

```diff
@@ -4594,6 +4594,97 @@ bool ValueNumStore::IsVNConstant(ValueNum vn)
     }
 }
 
+bool ValueNumStore::IsVNVectorZero(ValueNum vn)
+{
+    if (vn == NoVN)
+    {
+        return false;
+    }
+#if FEATURE_HW_INTRINSICS
+    // REVIEW: This will only return true if Vector.Zero encodes
+    //         its base type as an argument. On XARCH there may be
+    //         scenarios where Vector.Zero will not encode its base type;
+    //         therefore, returning false here.
+    VNFuncApp funcApp;
+    if (GetVNFunc(vn, &funcApp) && funcApp.m_arity == 1)
+    {
+        switch (funcApp.m_func)
+        {
+            case VNF_HWI_Vector128_get_Zero:
+#if defined(TARGET_XARCH)
+            case VNF_HWI_Vector256_get_Zero:
+#elif defined(TARGET_ARM64)
+            case VNF_HWI_Vector64_get_Zero:
+#endif
+            {
+                return true;
+            }
+
+            default:
+                return false;
+        }
+    }
+#endif
+    return false;
+}
+
+#if FEATURE_SIMD
+VNSimdTypeInfo ValueNumStore::GetSimdTypeOfVN(ValueNum vn)
+{
+    VNSimdTypeInfo vnInfo;
+
+    if (vn == NoVN)
+    {
+        vnInfo.m_simdSize        = 0;
+        vnInfo.m_simdBaseJitType = CORINFO_TYPE_UNDEF;
+        return vnInfo;
+    }
+
```

Comment:
```suggestion
```

No need for this case, it should be handled by the case below.

Reply:
That's true since getting the func app can return false.